collector/src/main/java/io/prometheus/jmx/logger/LoggerFactory.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (C) 2023 The Prometheus jmx_exporter Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.prometheus.jmx.logger;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/** Class to implement a LoggerFactory */
+public class LoggerFactory {
+
+    private static Map<Class<?>, Logger> LOGGER_MAP = Collections.synchronizedMap(new HashMap<>());
```

Comment:
This map is not needed, because you create only one `static final` instance per class anyway.

Reply:
Removed Map.